ire-adapter/src/main/scala/ingraph/ire/EngineFactory.scala

```diff
@@ -70,14 +70,14 @@ object EngineFactory {
             }
             remaining += ForwardConnection(op.child, node)
 
-          case op: BinaryFNode =>
+          case op: BinaryTNode =>
             val node: ActorRef = op match {
-              case op: Union => newLocal(Props(new UnionNode(expr.child, op.jnode.bag)))
+              case op: Union => newLocal(Props(new UnionNode(expr.child, op.fnode.jnode.bag)))
               case op: JoinLike =>
                 val child = expr.child
-                val leftTupleWidth = op.left.internalSchema.length
-                val rightTupleWidth = op.right.internalSchema.length
-                val leftMask = op.leftMask
+                val leftTupleWidth  = op.left. fnode.internalSchema.length
```

Comment:
whats up with the whitespace here?

Reply:
well, left and right are of different lengths :-)
I even considered moving to `pri` and `sec`, but let's leave that to the Rete network